src/main/kotlin/com/springernature/newversion/BuildpackVersionChecker.kt

```diff
@@ -12,6 +12,13 @@ data class SuccessfulChecks(val updates: List<BuildpackUpdate>) : ChecksResult()
 data class FailedChecks(val updates: List<BuildpackUpdate>, val errors: Map<BuildpackUpdate, Exception>) :
     ChecksResult()
 
+fun ChecksResult.exitStatus(): Int {
+    return when (this) {
+        is FailedChecks -> 1
+        is SuccessfulChecks -> 0
+    }
+}
+
```

Comment:
Just to be really picky, should this be abstract on `ChecksResult` and implemented appropriately by the two subclasses?

Reply:
I was wondering the same! :laughing: 
With my _very_ limited experience with Kotlin an "extension function" felt to be more "The Kotlin way", but … :shrug: 
I'm going to change it.

Reply:
Open question 😄  Myself, I prefer extension functions for additional or scoped functionality, and to stick to an OO model for general object members.

Something to argue out in general in #kotlin, perhaps!